src/diffTool/jsonDiff.ts

```diff
@@ -0,0 +1,374 @@
+/*
+ * Copyright (c) 2020, salesforce.com, inc.
+ * All rights reserved.
+ * SPDX-License-Identifier: BSD-3-Clause
+ * For full license text, see the LICENSE file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+ */
+/* eslint-disable @typescript-eslint/no-use-before-define */
+import _ from "lodash";
+import { DiffPatcher } from "jsondiffpatch";
+import { ramlToolLogger } from "../logger";
+
+/**
+ * Top level keys/property names in flattened JSON-LD
+ */
+export const JSON_LD_KEY_GRAPH = "@graph";
+export const JSON_LD_KEY_CONTEXT = "@context";
+export const JSON_LD_KEY_ID = "@id";
+
+/**
+ * External property, added by jsondiffpath to indicate that the node property is an array
+ */
+const JSON_DIFF_PATCH_ARRAY_KEY = "_t";
+const JSON_DIFF_PATCH_ARRAY_VALUE = "a";
+
+/**
+ * Difference types that can be identified by jsondiffpath
+ */
+export enum DiffType {
+  ADDED,
+  REMOVED,
+  MODIFIED,
+  TEXT_DIFF,
+  MOVED
+}
+
+/**
+ * Class to hold differences of a JSON node
+ */
+export class NodeDiff {
+  public added?: { [key: string]: any };
```

Comment:
It is optional as a property can be just removed or just added. I am initializing in the constructor  so that I  don't have to check every time I add a property. Also I am not sure at this point what is best to use with rules engine - an empty object or an undefined. I will clean this when I implement the rules  engine